snorkel/labeling/model/label_model.py

```diff
@@ -876,15 +904,41 @@ def fit(
             )
 
         self._set_constants(L_shift)
-        self._set_class_balance(class_balance, Y_dev)
-        self._create_tree()
+        self._common_training_preamble(
+            class_balance=class_balance, Y_dev=Y_dev, **kwargs
+        )
         lf_analysis = LFAnalysis(L_train)
         self.coverage = lf_analysis.lf_coverages()
 
         # Compute O and initialize params
         if self.config.verbose:  # pragma: no cover
             logging.info("Computing O...")
         self._generate_O(L_shift)
+        self._common_training_loop()
+
+    def _common_training_preamble(
```

Comment:
Nit: we can drop the `_common` prefix here and for the training loop method below, since the base class doesn't necessarily need to be aware that it has children that will be using these methods.